src/datahike/api.cljc

```diff
@@ -196,12 +197,18 @@
   transact
   (fn [connection arg-map]
     (let [arg (cond
-                (and (map? arg-map) (contains? arg-map :tx-data)) arg-map
-                (vector? arg-map)                                 {:tx-data arg-map}
-                (seq? arg-map)                                    {:tx-data arg-map}
-                :else                                             (dt/raise "Bad argument to transact, expected map with :tx-data as key.
-                               Vector and sequence are allowed as argument but deprecated."
-                                                                            {:error :transact/syntax :argument arg-map}))]
+                (or (vector? arg-map)
+                    (seq? arg-map))   (do (log/warn "Vector and sequence as argument are deprecated."
```

Comment:
I find `warn` a bit stark, `info` or `debug` should be fine. We should only do this runtime warning if we really have to change behaviour. If vector and sequence arguments are supposed to be used for something else in the future then it would be reasonable to do so, but are they? I personally find it compact to just pass a sequence of maps or Datoms, in particular if you transact something tiny.

Reply:
Ok, I have not made the decision to deprecate it. I guess it is best to ask @kordano and either send the message or keep it as is.

Reply:
I talked to @kordano and we can keep the support for vector and sequence arguments, so I removed the deprecation message.